packages/backend/src/projectAdapters/dbtLocalCredentialsProjectAdapter.ts

```diff
@@ -0,0 +1,38 @@
+import { CreateWarehouseCredentials } from 'common';
+import tempy from 'tempy';
+import * as path from 'path';
+import { writeFileSync } from 'fs';
+import {
+    LIGHTDASH_PROFILE_NAME,
+    LIGHTDASH_TARGET_NAME,
+    profileFromCredentials,
+} from '../dbt/profiles';
+import { DbtLocalProjectAdapter } from './dbtLocalProjectAdapter';
+
+type DbtLocalCredentialsProjectAdapterArgs = {
+    projectDir: string;
+    warehouseCredentials: CreateWarehouseCredentials;
+    port: number;
+};
+
+export class DbtLocalCredentialsProjectAdapter extends DbtLocalProjectAdapter {
+    constructor({
+        projectDir,
+        warehouseCredentials,
+        port,
+    }: DbtLocalCredentialsProjectAdapterArgs) {
+        const profilesDir = tempy.directory();
+        const profilesFilename = path.join(profilesDir, 'profiles.yml');
+        const { profile, environment } =
+            profileFromCredentials(warehouseCredentials);
+        writeFileSync(profilesFilename, profile);
```

Comment:
Does `tempy.directory();` generate a new directory each time ? otherwise we might be overriding the profiles.yml each time we test an adapter.

Reply:
And should we clean up the temporary directory on destroy()?

Reply:
Ah yes! I forgot to clean it up!

The plan here is to remove as much state from the adapter as possible, so I'd rather just recreate the profiles.yml each time rather than check it exists and then update it (and possibly hit a bug if it's been deleted).

But to save resources, I have the `destroy()` method which should be called most of the time to clean anything up.